pysite/oauth.py

```diff
@@ -0,0 +1,84 @@
+import logging
+from json import JSONDecodeError
+from uuid import uuid4, uuid5
+
+from flask import session
+from flask_dance.consumer.backend import BaseBackend
+from flask_dance.contrib.discord import discord
+import requests
+
+
+from pysite.constants import DISCORD_API_ENDPOINT, SERVER_ID
+
+
+class OauthBackend(BaseBackend):
+    """
+    This is the backend for the oauth
+
+    This is used to manage users that have completed
+    an oauth dance. It contains 3 functions, get, set,
+    and delete, however we only use set.
+
+    Inherits:
+        flake_dance.consumer.backend.BaseBackend
+        pysite.mixins.DBmixin
+
+    Properties:
+        key: The app's secret, we use it too make session IDs
+    """
+
+    def __init__(self, manager):
+        super(BaseBackend, self).__init__()
+        self.db = manager.db
+        self.key = manager.app.secret_key
+
+    def get(self, *args, **kwargs):  # Not used
+        pass
+
+    def set(self, blueprint, token):
+
+        user = self.get_user()
+        self.join_discord(token["access_token"], user["id"])
```

Comment:
Is there any way we can detect that we needed to do this and let the user know on the site?